### PR TITLE
Decrement cache tallies for uncached resources

### PR DESCRIPTION
**Pull request by zzmp**
Fixes `ResourceCache` tallies to account for resources created through the cache but marked as not cacheable. Although they are created through the cache, they go out of scope without being added to the cache, so the cache was not accounting for them.

### CODE REVIEW

**Comment by zzmp** [comment]
#### Testing

Open cacheStats.js.
Go to a domain with lots to load.
Go to another domain before loading is complete.

Before, objects in mid-load would stick in the stats.
Now, they should clear.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 32e9c49cf0ab2ce1d9d1fbed63cfc8978608d75e available at: https://gist.github.com/589b258f03ab8dd86159e38205841820<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.dmg

File: libraries/networking/src/ResourceCache.cpp [hunk 1]
```diff
@@ -196,11 +195,8 @@ void ResourceCache::addUnusedResource(const QSharedPointer<Resource>& resource)
     // If it doesn't fit or its size is unknown, remove it from the cache.
     if (resource->getBytes() == 0 || resource->getBytes() > _unusedResourcesMaxSize) {
         resource->setCache(nullptr);
-
-        _totalResourcesSize -= resource->getBytes();
-        _resources.remove(resource->getURL());
+        removeResource(resource->getURL(), resource->getBytes());
```
Comment (zzmp) [inline libraries/networking/src/ResourceCache.cpp]: This was actually missing the lock before.

**Comment by zzmp** [comment]
@ZappoMan I merged current master in. #7650 added a mutex to guard `_resources`, so I think it needs a re-review.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fc9de4b7e6b8d2593320bc42b12384dbf086a1c5 available at: https://gist.github.com/589b258f03ab8dd86159e38205841820<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.dmg

**Comment by kevinmthomas-carpool** [comment]
## QA Test
**PR Build:** 7673
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** localhost, playa, cellscience, forum
**Smoke Test Results:** Successful test.

                                  |localhost |playa    |cellscience |forum  
----------------------------------|----------|---------|------------|-----
Render Rate:                      |54.10     |54.08    |51.69       |52.73    
Present Rate:                     |60.04     |60.04    |60.04       |60.04   
Simrate:                          |52.10     |52.12    |50.20       |55.12    
Render Stats Num Buffers:         |1353      |3304     |2546        |4299   
Render Stats gpu::Buffer Memory:  |46.5Mb    |70.4Mb   |42.6Mb      |58.7Mb        
Render Stats Num Textures:        |351       |487      |197         |821    
Render Stats gpu::Texture Memory: |991.8Mb   |1170.3Mb |528.8Mb     |1521.2Mb         
Render Stats Triangles:           |409k      |590k     |574k        |1053k    
Render Stats Drawcalls:           |181       |286      |210         |898   
Render Stats Items:               |97        |246      |821         |381
GPU-Z Memory Size                 |4096Mb    |4096Mb   |4096Mb      |4096Mb
GPU-Z Initial Memory		  |239Mb     |239Mb    |239Mb       |239Mb
GPU-Z Memory Used		  |1127Mb    |969Mb    |849Mb       |1771Mb
GPU-Z Memory Load		  |13%       |12%      |9%          |10%

**localhost ‘createtestblocks.js’ Result:** Successful test.
**localhost ATP Upload (kelectricguitar10.fbx):** Successful test.
**localhost ‘harmonicOscillator.js’ Result:** Successful test.
**localhost Echo Server Audio Result:** Partially successful test.  When testing server audio on ```localhost```, it did not successfully echo back, upon visiting ```playa```, worked successfully upon revisiting ```localhost``` successful test.  Unsure if this is a real issue but wanted to bring it up for potential further analysis.  Enclosed is the log for further inspection.
**localhost ‘reverbTest.js’ Result:** Successful test.
**Additional Test Results Requested:** Successful test.  Reviewing ```cacheStats.js``` and upon visiting ```playa```, went to ```cellscience``` while ```playa``` was still loading and appears objects are no longer sticking in the stats as ```Texture Size``` and ```Texture Count```, both ```total``` and ```cached``` values appeared to clear out successfully throughout when moving domain to domain without allowing content to fully rez.
**Successful Logout/Crash:** Successful logout.
[attachments.zip](https://github.com/highfidelity/hifi/files/221351/attachments.zip)

**Comment by zzmp** [comment]
Oops...I clicked the wrong button. Didn't mean to close it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fc9de4b7e6b8d2593320bc42b12384dbf086a1c5 available at: https://gist.github.com/589b258f03ab8dd86159e38205841820<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.dmg

File: libraries/networking/src/ResourceCache.cpp [hunk 2]
```diff
@@ -259,6 +254,12 @@ void ResourceCache::resetResourceCounters() {
     emit dirty();
 }
 
+void ResourceCache::removeResource(const QUrl& url, qint64 size) {
+    QWriteLocker locker(&_unusedResourcesLock);
```
Comment (ZappoMan) [inline libraries/networking/src/ResourceCache.cpp]: wait - this is using  `_unusedResourcesLock`??? Is that intentional? don't you mean `_resourcesLock `

**Comment by hifi-gustavo** [comment]
Build console output for commit: cff5886313156c97bc3a297283194fbc10cee811 available at: https://gist.github.com/589b258f03ab8dd86159e38205841820<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7673.dmg

**Comment by cozza13** [comment]
Went through the test process. QA Approved.

**Pull request merged**